JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/timeseries/TimeSeriesClient.java

```diff
@@ -86,43 +92,43 @@ public void setKBClient(TripleStoreClientInterface kbClient) {
-	}
-
-	public void initInstantaneousTimeSeries(List<String> dataIRIs, List<Class<?>> dataClass, String timeUnit, Connection conn){
-		String tsIRI = TimeSeriesSparql.Instantaneous + "Timeseries_" + UUID.randomUUID();
-		initTimeSeries(tsIRI, dataIRIs, dataClass, timeUnit, conn, TimeSeriesSparql.InstantaneousTimeSeries, null, null);
-	}
-
-	public void initStepwiseCumulativeTimeSeries(List<String> dataIRIs, List<Class<?>> dataClass, String timeUnit, Connection conn){
-		String tsIRI = TimeSeriesSparql.StepwiseCumulative + "Timeseries_" + UUID.randomUUID();
-		initTimeSeries(tsIRI, dataIRIs, dataClass, timeUnit, conn, TimeSeriesSparql.StepwiseCumulativeTimeSeries, null, null);
-	}
-
-	public void initCumulativeTotalTimeSeries(List<String> dataIRIs, List<Class<?>> dataClass, String timeUnit, Connection conn){
-		String tsIRI = TimeSeriesSparql.CumulativeTotal+  "Timeseries_" + UUID.randomUUID();
-		initTimeSeries(tsIRI, dataIRIs, dataClass, timeUnit, conn, TimeSeriesSparql.CumulativeTotalTimeSeries, null, null);
-	}
-
-	public void initGeneralTimeSeries(List<String> dataIRIs, List<Class<?>> dataClass, String timeUnit, Connection conn){
-		String tsIRI = TimeSeriesSparql.ns_kb + "Timeseries_" + UUID.randomUUID();
-		initTimeSeries(tsIRI, dataIRIs, dataClass, timeUnit, conn, TimeSeriesSparql.TimeSeries, null, null);
-	}
-
-	private void initTimeSeries(String tsIRI, List<String> dataIRIs, List<Class<?>> dataClass, String timeUnit, Connection conn, Iri type, Duration duration, ChronoUnit unit) {
+	public void initTimeSeries(List<String> dataIRIs, List<Class<?>> dataClass, String timeUnit, Connection conn, Type type, Duration duration, ChronoUnit unit) {
 
 		// Step1: Initialise time series in knowledge base
 		// In case any exception occurs, nothing will be created in kb, since JPSRuntimeException will be thrown before
 		// interacting with triple store and SPARQL query is either executed fully or not at all (no partial execution possible)
+		String tsIRI;
+		Iri timeseriesType;
+
+		if(type.equals(Type.AVERAGE)){
+			timeseriesType = TimeSeriesSparql.AverageTimeSeries;
+			tsIRI = TimeSeriesSparql.Average + "Timeseries_" + UUID.randomUUID();
+		}
+		else if(type.equals(Type.STEPWISECUMULATIVE)){
+			timeseriesType = TimeSeriesSparql.StepwiseCumulativeTimeSeries;
+			tsIRI = TimeSeriesSparql.StepwiseCumulative + "Timeseries_" + UUID.randomUUID();
+		}
+		else if(type.equals(Type.CUMULATIVETOTAL)){
+			timeseriesType = TimeSeriesSparql.CumulativeTotalTimeSeries;
+			tsIRI = TimeSeriesSparql.CumulativeTotal+  "Timeseries_" + UUID.randomUUID();
+		}
+		else if(type.equals(Type.INSTANTANEOUS)){
+			timeseriesType = TimeSeriesSparql.InstantaneousTimeSeries;
+			tsIRI = TimeSeriesSparql.Instantaneous + "Timeseries_" + UUID.randomUUID();
+		}
+		else {
+			throw new JPSRuntimeException(exceptionPrefix + "TimeSeries type: " + type + " is invalid");
```

Comment:
You don't need this, it's impossible to define an invalid type with enum

Reply:
removed it